pkg/apis/controller/v1beta1/appwrapper.go

```diff
@@ -83,19 +100,27 @@ type AppWrapperResource struct {
 	MinAvailable *int32 `json:"minavailable,omitempty" protobuf:"bytes,3,opt,name=minavailable"`
 
 	// The number of allocated replicas from this resource type
+	// +optional
 	AllocatedReplicas int32 `json:"allocatedreplicas"`
 
-	// The priority of this resource
-	Priority float64 `json:"priority"`
+	// +kubebuilder:validation:Type=number
+	// +kubebuilder:validation:Format=int
+	// +optional
+	Priority float64 `json:"priority,omitempty"`
```

Comment:
`kubebuilder:validation:Format=int` mismatch with `Priority float64`.  I suspect we have a missmatch in the schema's as well.  Should stay as float64 and change format as `float`